script.service.singealarmclock/plugin.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+#     Copyright (C) 2012 Team-XBMC
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#    This script is based on service.skin.widgets
+#    Thanks to the original authors
+
+import xbmc, xbmcaddon
+import os, sys
+__cwd__      = xbmc.translatePath( xbmcaddon.Addon().getAddonInfo('path') ).decode("utf-8")
+__resource__ = xbmc.translatePath( os.path.join( __cwd__, 'resources', 'lib' ) ).decode("utf-8")
+sys.path.append(__resource__)
```

Comment:
bad practice to use sys.path.append. Why do you need it? Can't you just import your python code as modules?